From 88bb62ee65721e2591f989e2bda3e1ab08efdf2b Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 15 Dec 2022 17:15:49 +0800
Subject: [PATCH 5/6] wifi: mt76: mt7915: set sku initial value to zero

Set sku initial value to prevent mcu get an invalid value before mcu starts
to run.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7915/soc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mt7915/soc.c b/mt7915/soc.c
index 686c9bb..2ac0a0f 100644
--- a/mt7915/soc.c
+++ b/mt7915/soc.c
@@ -883,6 +883,8 @@ static int mt7986_wmac_wm_enable(struct mt7915_dev *dev, bool enable)
 {
 	u32 cur;
 
+	mt76_wr(dev, MT_CONNINFRA_SKU_DEC_ADDR, 0);
+
 	mt76_rmw_field(dev, MT7986_TOP_WM_RESET,
 		       MT7986_TOP_WM_RESET_MASK, enable);
 	if (!enable)
-- 
2.36.1

